### DISCUSSION TITLE
Add the ability to Add New in m2o relation field

### DISCUSSION THREAD

**Discussion by lluishi93** [directus/directus#2742] [Feature Requests]
The same Add new button that appears in the m2m fields, would be nice to have it in the m2o fields as well, to create entries directly and improve the UX a bit.

- Button in m2m

![image](https://user-images.githubusercontent.com/14786440/83657032-99271280-a5c0-11ea-8cc4-b962e0be6c53.png)

- M2o field without the button

![image](https://user-images.githubusercontent.com/14786440/83657059-a6440180-a5c0-11ea-8675-6119ef8f4aca.png)

After the creation of the item, it would be nice to automatically relate it to the one that is being edited.

**Comment by benhaynes** [comment]
Definitely. This is part of our big focus on standardization and reusing components in v9. Below is a mock of how this might look... notice we could go with the icon, or the dropdown menu approach for adding new items. Thoughts?

![Screen Shot 2020-06-03 at 1 13 49 PM](https://user-images.githubusercontent.com/522079/83667310-2c018600-a59c-11ea-9b05-bfc29decd1cb.png)

**Reply by yasanovich** [reply]
From your answer, it is not entirely clear whether it will be possible to add a new O2M item directly from the collection with the M2O field.

**Reply by rijkvanzanten** [reply]
@yasanovich It says _Create a new item_ right there in the dropdown, and the other many to ones all feature a `+` that's visible in the designs

**Reply by yasanovich** [reply]
@rijkvanzanten Oh, thanks, it’s strange that I didn’t conceal it. 
@benhaynes Then the most logical question is when will it be?

**Reply by benhaynes** [reply]
Haha, yeah... for us too! 😉 

The timeline for v9 would have been earlier, but got pushed when we decided to go big for this release (also adding a node API port and redoing our marketing site). No official release date, but hopefully in about a month or so.

**Reply by benhaynes** [reply]
@rijkvanzanten — did you add this with the others?

**Reply by rijkvanzanten** [reply]
Yes

**Comment by lluishi93** [comment]
@benhaynes Both of them look nice to me! :) Great improvement IMHO.

**Comment by LasseRosenow** [comment]
Why is this marked as implemented? I cannot find any way to do this for M2O in the latest release

**Reply by rijkvanzanten** [reply]
<img width="463" alt="CleanShot 2021-08-10 at 14 39 41@2x" src="https://user-images.githubusercontent.com/9141017/128916509-e424c53a-1180-4f65-a160-04cb23ef9c01.png">

**Reply by LasseRosenow** [reply]
Oh thank you very much, I totally missed that. Maybe it would be great to make that more consistent with the others by switching to a green button aswell.

Also being able to disable the ability to "select" something would be nice I think

Also having a green + icon in the select overview would be nice to make things more accessible maybe
